solar_system.rb

```diff
@@ -0,0 +1,262 @@
+    end
+    return list
+  end
+
+  def one_planet(user_planet)
+    planet_details = ""
+    @planets.each_with_index do |planet, i|
+      if user_planet == planet.name
+        planet_details << "---\nName: #{planet.name}\nColor: #{planet.color}\nShape: #{planet.shape}\nMotto: #{planet.motto}\nPlanetary bird: #{planet.planetary_bird}\nYear length (in days): #{planet.year_length}\nDistance from the sun (in feet): #{planet.distance_from_the_sun}\n---\n"
+      end
+    end
+
+    return planet_details
+  end
+
+  # Ensure that the each planet has a @distance_from_the_sun attribute. Using this data, add a method to determine the distance from any other planet (assuming planets are in a straight line from the sun)
+  def distance_btwn_planets(planet_1, planet_2)
+    # STILL TO DO - Verify that the planets are different
+
+    # find planet with larger distance
+    # subtract smaller distance from larger distance
+    # return result
+
+    #  a <=> b
+      # if a < b then return -1
+      # if a = b then return  0
+      # if a > b then return  1
+
+    if (planet_1.distance_from_the_sun <=> planet_2.distance_from_the_sun) == 1
+      distance = planet_1.distance_from_the_sun - planet_2.distance_from_the_sun
+      return distance
+    elsif (planet_1.distance_from_the_sun <=> planet_2.distance_from_the_sun) == -1
+      distance = planet_2.distance_from_the_sun = planet_1.distance_from_the_sun
+      return distance
+    else # they are equal
+      puts "You entered in the same planet twice! It's distance from itself is 0."
+    end
+  end
+
+  # Define a method that returns the local year (earth years) of the planet based on it's rotation since the beginning of the solar system
+
+  # earth = x * 365/365 (x = age of solar system)
+  # jupiter = x * 500/365
+
+  def earth_years
+    earth_year_list = ""
+    @planets.each_with_index do |planet, i|
+      # earth_year_list << @age * (planet.year_length / 365)
+        earth_year_list << "---\nName: #{planet.name}\nAge in earth years: #{@age * (planet.year_length / 365.0)}\n"
+      end
+    return earth_year_list
+  end
+
+end
+
+# Create instances of planets
+planet_1 = Planets.new("Winnicott", "Technicolour", "Heart", "You are good", "Cedar waxwing", 500, 2323232)
+planet_2 = Planets.new("Rollo", "Black", "Round", "Deep breath", "Great blue heron", 23, 100)
+planet_3 = Planets.new("Yalom", "Bright yellow", "Square", "Star at the sun", "Emperor penguin", 123, 10)
+planet_4 = Planets.new("KENN", "Red", "Sphere", "Connections are everywhere", "American Robin", 945, 250)
+planet_5 = Planets.new("Earth", "Multicolour", "Sphere", "???", "All birds", 365, 1000)
+
+# Create instance of solar system using previous instances of planets
+my_system = SolarSystem.new([planet_1, planet_2, planet_3, planet_4, planet_5])
```

Comment:
Since you're creating all these local variables just to add them to an array, a more concise way to do this might be:

```ruby
planet_array = [
  Planets.new("Winnicott", "Technicolour", "Heart", "You are good", "Cedar waxwing", 500, 2323232),
  Planets.new("Rollo", "Black", "Round", "Deep breath", "Great blue heron", 23, 100),
  # ...
]
my_system = SolarSystem.new(planet_array)
```

This makes it clear to the reader that we won't be using the planets individually, only through the collection, and also saves you having to come up with a bunch of variable names.